lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -3578,4 +3581,30 @@ public function getStringLiteralQuoteCharacter()
     {
         return "'";
     }
+
+    /**
+     * Escapes metacharacters in a string intended to be used with a LIKE
+     * operator.
+     *
+     * @param string $inputString a literal, unquoted string
+     * @param string $escapeChar  should be reused by the caller in the LIKE
+     *                            expression.
+     */
+    final public function escapeStringForLike(string $inputString, string $escapeChar) : string
+    {
+        $replacePairs = [$escapeChar => $escapeChar . $escapeChar];
+        foreach ($this->getLikeWildcardCharacters() as $wildcardChar) {
+            $replacePairs[$wildcardChar] = $escapeChar . $wildcardChar;
+        }
+
+        return strtr($inputString, $replacePairs);
+    }
+
+    /**
+     * @return string[]
+     */
+    protected function getLikeWildcardCharacters() : array
+    {
+        return ['%', '_'];
```

Comment:
Does this really need to be an array? Why not a string?

Reply:
Didn't think of it, but that would enforce the characters being characters